### DISCUSSION TITLE
What is the intended behaviour of "Return to Stock"?

### DISCUSSION THREAD

**Discussion by miggland** [inventree/InvenTree#4200] [Q&A] [answered]
We recently sold some stock to a customer. I now had to recall it, because we need to perform a repair on it.

I went to the stock item in Inventree and used the "Return to Stock" function:
![image](https://user-images.githubusercontent.com/1314853/211813401-1c3d716c-6403-41fa-9949-aaf8c2f80c79.png)

After that, it shows up as "Assigned to sales order" in the stock list for the respective part. It's the second line. The third line is the same part, sent separately, and not returned:
![image](https://user-images.githubusercontent.com/1314853/211813724-adbdc82a-b435-4ffa-bc7e-fc15d8c6378f.png)

Now, if I filter the stock list for available items I get the following:
![image](https://user-images.githubusercontent.com/1314853/211813938-6146a9fd-d64d-4e71-8440-91ba3fdda3e0.png)

The questions is, where is the returned stock item? It's not marked as available, but is still assigned (in some views). In the stock history, the change listed is a return to the given stock location.

What's the intended behaviour here? Is the part not supposed to be marked as available when returned? How would I go about marking the part as "Available" again?

**Comment by matmair** [comment, marked as answer]
@miggland the part should be marked as available just by the action. Seems to be a bug.
Could you try to reproduce this behaviour on the demo site and open a bug report?
